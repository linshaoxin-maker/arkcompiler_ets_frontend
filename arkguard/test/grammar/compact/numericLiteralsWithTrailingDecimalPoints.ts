/*
 * Copyright (c) 2023 Huawei Device Co., Ltd.
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

1..toString();
1.0.toString();
1. + 2.0 + 3.;

var i: number = 1;
var case1 = i.toString();
var case3 = 3 .toString();
var case4 = 3    .toString();
var case5 = 3	.toString();
var case6 = 3.['toString']();
var case7 = 3
  .toString();
var case8 = new Number(4).toString();
var case9 = 3. + 3.;
var case10 = 0 /* comment */.toString();
var case11 = 3. /* comment */.toString();
var case12 = 3
  /* comment */ .toString();
var case122 = 3
/* comment */.toString();
var case1222 = 3

  .toString();
var case13 = 3.
  /* comment */.toString();
var case14 = 3
  // comment
  .toString();
var case15 = 3.
  // comment
  .toString();
var case16 = 3  // comment time
  .toString();
var case17 = 3. // comment time again
  .toString();




