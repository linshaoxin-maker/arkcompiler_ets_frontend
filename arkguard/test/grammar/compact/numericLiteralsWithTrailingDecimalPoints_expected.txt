                           1..toString(); 1.0.toString(); 1. + 2.0 + 3.; var n: number = 1; var o = n.toString(); var p = 3..toString(); var q = 3..toString(); var r = 3..toString(); var s = 3.['toString'](); var t = 3 .toString(); var u = new Number(4).toString(); var v = 3. + 3.; var w = 0 .toString(); var x = 3. .toString(); var y = 3   .toString(); var z = 3   .toString(); var a1 = 3 .toString(); var b1 = 3.   .toString(); var c1 = 3   .toString(); var d1 = 3.   .toString(); var e1 = 3   .toString(); var f1 = 3.   .toString(); 